### PR TITLE
migrate cluster_drs_recommendations module

### PR DESCRIPTION
**Pull request by mikemorency**
##### SUMMARY
This migrates the cluster_drs_recommendations module from the community.vmware collection

The inputs have not changed.
The outputs have changed from the original module. This version reports a description of the applied recommendations as well as details about the vcenter task associated with the change

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
cluster_drs_recommendations

### CODE REVIEW

**Comment by anna-savina** [comment]
@mikemorency what do you think about next questions?

1. Is there an alternative approach to testing this module? 
Currently, the test cluster is empty, and there are no recommendations, meaning the test doesn't actually perform anything.

2. What should the result be if there are no recommendations? 
The `community.vmware.vmware_cluster_drs_recommendations` module records in the results that no changes occurred
```
    "recommendations_info": {
        "changed": false,
        "failed": false,
        "result": "No recommendations."
    }
```

while `vmware.vmware.cluster_drs_recommendations` shows that changes were made
```
    "recommendations_info": {
        "applied_recommendations": [],
        "changed": true,
        "failed": false
    }
```

**Comment by mikemorency** [comment]
@anna-savina 
1. Validating that recommendations are made and run requires a vcenter with multiple hosts running multiple VMs in an unbalanced way. I think setting up that testing scenario will take awhile. I think that might be too long for the PR tests we run, what do you think?
I was going to mention this in our QE sync but maybe we could have a longer test that runs nightly or something. That could stand up an entire test vcenter and then fully test this and other modules. Or maybe we could create some test VM templates (esxi, rhel9, etc) so spinning up a new complete cluster is much faster?

2. Your right, the module should not report that something has changed if theres no recommendations. Ill fix that

**Comment by anna-savina** [comment]
> I think setting up that testing scenario will take awhile. I think that might be too long for the PR tests we run, what do you think?

yes, I agree with you. I didn't find right approach to test this module, thought maybe you have ideas.

> I was going to mention this in our QE sync but maybe we could have a longer test that runs nightly or something.

I'm not sure about this. Let's discuss it during our QE sync meeting

**Review by bardielle — APPROVED** [review]

File: plugins/modules/cluster_drs_recommendations.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+EXAMPLES = r'''
+- name: Apply DRS Recommendations for Cluster
+  vmware.vmware.cluster_drs_recommendations:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datacenter_name: datacenter
+    cluster_name: cluster
+    enable: true
```
Comment (anna-savina) [inline plugins/modules/cluster_drs_recommendations.py:47]: Is 'enable' supported parameter?
`FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python3"}, "changed": false, "msg": "Unsupported parameters for (vmware.vmware.cluster_drs_recommendations) module: enable. Supported parameters include: cluster, datacenter, hostname, password, port, proxy_host, proxy_port, username, validate_certs (admin, cluster_name, datacenter_name, pass, pwd, user)."}`

Comment (mikemorency) [reply plugins/modules/cluster_drs_recommendations.py:47]: Nope, i missed that in the example docs. Ill remove it. I dont see it anywhere else (plugin code, tests, etc), let me know if you do

**Review by anna-savina — APPROVED** [review]
LGTM

**Pull request merged**